### PR TITLE
Adds functionality to support shortcuts for adding all members of a g…

### PR DESCRIPTION
**Pull request by dittonjs**
…roup or section to a conference.

This change adds checkboxes for groups and sections to the members list section of the dialog for creating / editing a conference. These checkboxes are "shortcuts" that allow the user to quickly add all members of a specific section or group to a conference. These shortcuts are implemented client side and the  model for conferences is unchanged.

Test Plan:
	- Make sure the conferences feature is enabled
	- In a course open the dialog to create a new conference
	- Uncheck the "Invite All" box
	- Verify the following in the member list:
		- A checkbox for each section should appear in the "Sections" partition (only if there is more than one section in the course)
		- A checkbox for each group should appear in the "Groups" partition (only if there is at least one section in the course)
		- The headers for "Sections" and "Groups" should only appear if there is atleast item in the respective partition
		- The header for "Users" should only appear if one or more of the other partitions is present
		- Checking the box for section / group also checks the boxes for each of the users in that section / group
		- Unchecking the box for section / group also unchecks the boxes for each of the users in that section / group provided that another section / group that contains that user isn't also checked
		- If a user is added to the conference via a section or group you should not be able to uncheck the checkbox for that user.

### CODE REVIEW

**Comment by ffdixon** [comment]
Here's a video demonstrating the feature in action.  

https://youtu.be/lzgkyH1lVHc

**Review by fadedmax — APPROVED** [review]
LGTM

**Comment by blunckr** [comment]
Hello, any idea when this might be merged? I'm not sure what your processes are.

**Comment by jbasdf** [comment]
Any update on this? Is there anything we can do to help get this merged in?

**Comment by simonista** [comment]
Hi, thanks for the ping and sorry for the slow response. Let me run this by the product manager for conferences and make sure this is functionality they want to add.

**Comment by ffdixon** [comment]
Thanks Simon.  This feature has been requested by a number large customers of Canvas Conferences and we sponsored the work to help Instructure get this feature in place.  We would like to see this go through the internal review process, support you however we can, make some customers very happy.

We'll fix the merge conflicts as well.

File: app/controllers/conferences_controller.rb [hunk 1]
```diff
@@ -278,6 +278,18 @@ def web_index(conferences)
       @render_alternatives = WebConference.conference_types(@context).all? { |ct| ct[:replace_with_alternatives] }
       case @context
       when Course
+        @sections = @context.course_sections
```
Comment (cguanzon) [inline app/controllers/conferences_controller.rb:281]: preload participants

File: app/controllers/conferences_controller.rb [hunk 2]
```diff
@@ -278,6 +278,18 @@ def web_index(conferences)
...
+        @groups = @context.active_groups
```
Comment (cguanzon) [inline app/controllers/conferences_controller.rb:282]: preload participating users in context

**Comment by cguanzon** [comment]
the following tests failed:

[2021-07-06T16:17:59.161Z] 1) sections should appear in member list if course has more than one section

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:204:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:205:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 2) groups should appear in member list if course has one or more groups

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:239:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 3) checking/unchecking a section also checks/unchecks the members that are in that section

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:257:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 4) checking/unchecking a groups also checks/unchecks the members that are in that group

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:276:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 5) unchecking a group only unchecks members that have not been selected by section also

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:296:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:297:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:299:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 6) unchecking a section only unchecks members that have not been selected by group also

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:317:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:318:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:320:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z] 7) While editing a conference the box for a group should be checked and disabled if everyone in the group is a participant

[2021-07-06T16:17:59.162Z]      EditConferenceView

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:339:3)

[2021-07-06T16:17:59.162Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.162Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.162Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.162Z] 

[2021-07-06T16:17:59.162Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.162Z] Expected: true

[2021-07-06T16:17:59.162Z] Actual: false

[2021-07-06T16:17:59.162Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.162Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:340:3)

[2021-07-06T16:17:59.163Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.163Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.163Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 8) While editing a conference the box for a section should be checked and disabled if everyone in the section is a participant

[2021-07-06T16:17:59.163Z]      EditConferenceView

[2021-07-06T16:17:59.163Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.163Z] Expected: true

[2021-07-06T16:17:59.163Z] Actual: false

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.163Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:358:3)

[2021-07-06T16:17:59.163Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.163Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.163Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.163Z] Expected: true

[2021-07-06T16:17:59.163Z] Actual: false

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.163Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:359:3)

[2021-07-06T16:17:59.163Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.163Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.163Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 9) While editing a conference unchecking a group should only uncheck members who are not a part of the existing conference

[2021-07-06T16:17:59.163Z]      EditConferenceView

[2021-07-06T16:17:59.163Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.163Z] Expected: true

[2021-07-06T16:17:59.163Z] Actual: false

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.163Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:381:3)

[2021-07-06T16:17:59.163Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.163Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.163Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 

[2021-07-06T16:17:59.163Z] 10) While editing a conference unchecking a section should only uncheck member who are not a part of the existing conference

[2021-07-06T16:17:59.163Z]      EditConferenceView

[2021-07-06T16:17:59.163Z]      failed, expected argument to be truthy, was: false

[2021-07-06T16:17:59.163Z] Expected: true

[2021-07-06T16:17:59.163Z] Actual: false

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:2194:12

[2021-07-06T16:17:59.163Z]     at Object.eval (webpack:///./spec/javascripts/jsx/conferences/EditConferenceViewSpec.js?:406:3)

[2021-07-06T16:17:59.163Z]     at runTest (node_modules/qunitjs/qunit/qunit.js:843:28)

[2021-07-06T16:17:59.163Z]     at Test.run (node_modules/qunitjs/qunit/qunit.js:828:4)

[2021-07-06T16:17:59.163Z]     at node_modules/qunitjs/qunit/qunit.js:970:11

[2021-07-06T16:17:59.163Z]     at process (node_modules/qunitjs/qunit/qunit.js:629:24)

**Review by cguanzon — CHANGES_REQUESTED** [review]
tests here https://github.com/instructure/canvas-lms/pull/1765/files#:~:text=spec/javascripts/jsx/conferences/EditConferenceViewSpec.js are failing for us

please check and make sure they pass

**Comment by bfcoder** [comment]
@cguanzon I'm not seeing the same failures.

```
  EditConferenceView
    ✓ updateConferenceUserSettingDetailsForConference localizes values for datepicker settings (45ms)
    ✓ #show sets the proper title for new conferences (119ms)
    ✓ #show sets the proper title for editing conferences (87ms)
    ✓ #show sets localized durataion when editing conference (93ms)
    ✓ "remove observers" modifies "invite all course members" (104ms)
    ✓ sections should appear in member list if course has more than one section (90ms)
    ✓ sections should not appear in member list if course has only section (81ms)
    ✓ groups should appear in member list if course has one or more groups (189ms)
    ✓ checking/unchecking a section also checks/unchecks the members that are in that section (91ms)
    ✓ checking/unchecking a groups also checks/unchecks the members that are in that group (91ms)
    ✓ unchecking a group only unchecks members that have not been selected by section also (89ms)
    ✓ unchecking a section only unchecks members that have not been selected by group also (87ms)
    ✓ While editing a conference the box for a group should be checked and disabled if everyone in the group is a participant (82ms)
    ✓ While editing a conference the box for a section should be checked and disabled if everyone in the section is a participant (87ms)
    ✓ While editing a conference unchecking a group should only uncheck members who are not a part of the existing conference (86ms)
    ✓ While editing a conference unchecking a section should only uncheck member who are not a part of the existing conference (91ms)
    ✓ while context_is_group = true no sections or groups should appear in the member list (91ms)

Chrome 92.0.4515 (Linux 0.0.0): Executed 17 of 17 SUCCESS (1.609 secs / 1.603 secs)
```

I'm running the specs with `yarn jspec-watch spec/javascripts/jsx/conferences/EditConferenceViewSpec.js`. How are you running them that you are seeing these failures?

Here are my versions:
```
yarn --version
1.22.5

node --version
v14.16.0
```

Thanks!

**Comment by roor0** [comment]
this was merged into canvas-lms 8a3c737b92b59721d5bc1c09a15aa32c8f0e63eb

**Pull request closed without merging**